manager/scheduler/filter.go

```diff
@@ -61,13 +61,34 @@ func (f *ResourceFilter) SetTask(t *api.Task) bool {
 	if r == nil || r.Reservations == nil {
 		return false
 	}
-	if r.Reservations.NanoCPUs == 0 && r.Reservations.MemoryBytes == 0 {
+
+	res := r.Reservations
+	if res.NanoCPUs == 0 && res.MemoryBytes == 0 && len(res.ThirdParty) == 0 {
 		return false
 	}
+
 	f.reservations = r.Reservations
 	return true
 }
 
+// HasEnoughThirdPartyResources returns true if node can satisfy task TPR's request
+func HasEnoughThirdPartyResources(nodeRes *api.ThirdPartyResource, taskRes *api.ThirdPartyResource) bool {
+	t := taskRes.Resource.(*api.ThirdPartyResource_Integer).Integer.Val
```

Comment:
This should use the `t, ok := ...` form to confirm that the resource reservation actually uses the integer type. While that's the way this is intended to be used, these protobuf messages are created by users, so there they can deviate from correct usage sometimes.